bin/run-amp-validator.js

```diff
@@ -0,0 +1,228 @@
+    console.log('----------------------------------------------------------------------------\n'.error);
+};
+
+/**
+ * Validate urls and add results to our* results variables
+ * @param   errors  array of error messages
+ */
+var validateUrls = function( urls ) {
+    const ourInstance = ampValidator.getInstance();
+    var i = 0,
+        len = urls.length - 1;
+
+    return promiseWhile(function() {
+        return i <= len;
+    }, function() {
+        return new Promise( function( resolve, reject ) {
+            var url = urls[i];
+            fetch( url )
+                .then( function( res ) {
+                    if ( res.ok ) {
+                        return res.text();
+                    } else {
+                        var response = 'FAIL: '.error + 'Unable to fetch ' + url + ' - HTTP Status ' + res.status + ' - ' + res.statusText + '\n';
+                        ourErrors.push( response );
+                        ourResults.push( 'FAIL' );
+                        i++;
+                        resolve();
+                    }
+                }).then(function(body) {
+                    if (body) {
+                        //We use timeout here because the AMP validator tool needs an internet connection.  If you are testing locally we want this to fail.  Without the timeout mocha was just failing assertion 1, but didn't provide any information as to why it failed.
+                        return timeout(2000,
+                            ourInstance)
+                            .then(function (validator) {
+                                const result = validator.validateString(body);
+                                if (result.status === 'PASS') {
+
+                                    ourResults.push('PASS');
+
+                                } else {
+                                    let msg = result.status.error + ": " + url + '\n';
+                                    for (const error of result.errors) {
+                                        msg += ('     line ' + error.line + ', col ' + error.col + ': ').debug + error.message.error;
+                                        if (error.specUrl !== null) {
+                                            msg += '\n     (see ' + error.specUrl + ')';
+                                        }
+                                    }
+                                    ourErrors.push(msg);
+                                    ourResults.push('FAIL');
+                                }
+                                i++;
+                                resolve();
+                            }).catch(function (reason) {
+                                i++;
+                                ourErrors.push(reason);
+                                console.error('Error or timeout', reason);
+                                reject(reason);
+                            });
```

Comment:
This has been better addressed with [this commit](https://github.com/iamchrissmith/amp-wp/commit/a43e140bca67b87b200fdb33fc3faca8d6ad6bef)